spot-client/src/common/auto-update/AutoUpdateChecker.js

```diff
@@ -0,0 +1,136 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { getUpdateEndHour, getUpdateStartHour } from 'common/app-state';
+import { logger } from 'common/logger';
+import { date } from 'common/utils';
+
+
+const checkInterval = 5 * 60 * 1000; // 5 minutes in milliseconds
+
+const lastLoadTime = new Date();
```

Comment:
somehow this does not feel very accurate because it's inside of a component which will be refreshed on navigation between views ? what we care about are page loads not component loads

Reply:
As you know Im not a big fan of smart component, but even if I try to like it then somehow this feels to fit more into a service which is not tied to a lifetime of a component (because of the issue mentioned in the comment above).

Reply:
I'm thinking that a component is fine to be plugged into the UI screen that supports the reload, but maybe it could work together with a service which will be accurate about tracking the last load times ?

Reply:
hmm wait I'm just stupid. This date will be when the page was loaded, because it is initialized when the code is loaded, right ?

Reply:
Right, that's why; it's essentially when the js bundle has been executed by the browser. I was thinking I'd make a model/service for this but forgot too. The modal can probably keep track of polling and use pub/sub, that way the little logic there is can be moved around a bit. I'll do that.

Reply:
No, wait. This is good for now.